crates/fluvio-spu/src/services/public/produce_handler.rs

```diff
@@ -154,7 +154,9 @@ fn validate_records<R: BatchRecords>(
         match compression {
             CompressionAlgorithm::Any => true,
             CompressionAlgorithm::None => batch_compression == Compression::None,
-            CompressionAlgorithm::Gzip => batch_compression == Compression::Gzip,
+            CompressionAlgorithm::Gzip => {
+                batch_compression == Compression::Gzip(GzipLevel::default())
```

Comment:
probably here, we could use `matches!( batch_compression, Compression::Gzip(_))`